server/routes/referrals/referralsController.ts

```diff
@@ -101,21 +101,26 @@ export default class ReferralsController {
     }
 
     if (error === null) {
-      const referral = await this.interventionsService.createDraftReferral(
-        res.locals.user.token.accessToken,
-        crn,
-        interventionId
-      )
-
-      await this.interventionsService.patchDraftReferral(res.locals.user.token.accessToken, referral.id, {
-        serviceUser: this.interventionsService.serializeDeliusServiceUser(serviceUser),
-      })
-
-      res.redirect(303, `/referrals/${referral.id}/form`)
+      const [referral, intervention] = await Promise.all([
+        this.interventionsService.createDraftReferral(res.locals.user.token.accessToken, crn, interventionId),
+        this.interventionsService.getIntervention(res.locals.user.token.accessToken, interventionId),
+      ])
+      if (intervention.serviceCategories.length === 1) {
+        await this.interventionsService.patchDraftReferral(res.locals.user.token.accessToken, referral.id, {
+          serviceUser: this.interventionsService.serializeDeliusServiceUser(serviceUser),
+          serviceCategoryIds: [intervention.serviceCategories[0].id],
+        })
+        res.redirect(303, `/referrals/${referral.id}/form`)
+      } else {
+        await this.interventionsService.patchDraftReferral(res.locals.user.token.accessToken, referral.id, {
```

Comment:
Just thinking out loud - if we're redirecting back to the form (see other comment), do you think it could be cleaner to do something like:
```
const serviceCategoryIds = intervention.ServiceCategories.length === 1 ? [intervention.serviceCategories[0].id] : null

await this.interventionsService.patchDraftReferral(res.locals.user.token.accessToken, referral.id, {
  serviceUser: this.interventionsService.serializeDeliusServiceUser(serviceUser),
  serviceCategoryIds: serviceCategoryIds,
})

res.redirect(303, `/referrals/${referral.id}/form`)
```

I guess we'd always be patching `serviceCategoryIds` here, do you think that'd have any performance issues (or other non-performance issues)?

Reply:
Yes, I think it was also because I was expecting the rediction to be different as well. I've used your suggestion.